packages/salesforce-adapter/test/client.test.ts

```diff
@@ -232,21 +237,14 @@ describe('salesforce client', () => {
   describe('queryAll', () => {
     let resultsIterable: AsyncIterable<Values[]>
     let dodoScope: nock.Scope
-    const asyncIterableValuesCounter = async (
-      iterator: AsyncIterator<Values[]>
-    ): Promise<number> => {
-      let counter = 0
-      const next = async (): Promise<void> => {
-        const curr = await iterator.next()
-        if (curr.done) return undefined
-        if (_.isArray(curr.value) && curr.value.length > 0) {
-          counter += curr.value.length
-        }
-        return next()
-      }
-      await next()
-      return counter
-    }
+
+    const asyncCounter = async (
+      iterator: AsyncIterable<Values[]>
+    ): Promise<number> =>
+      _.sum(_.flatten(await toArrayAsync(await mapAsync(
+        iterator,
+        (vals: Values[]) => makeArray(vals).map(_v => 1)
```

Comment:
nit: pretty sure the type can be inferred here
```suggestion
        vals => makeArray(vals).map(() => 1)
```